### DISCUSSION TITLE
[Chapter 10] 레디스 아키텍처와 CQRS 패턴

### DISCUSSION THREAD

**Discussion by corock** [cowaine/spring-boot-for-msa#155] [Chapter 10. Redis and Spring Cache] [answered]
[10.1 레디스 소개 및 아키텍처] 장에서는 다음과 같이 대표적인 레디스 아키텍처를 소개하고 있습니다.

1. 레디스 센티넬 아키텍처
2. 레디스 클러스터 아키텍처

아키텍처는 조금씩 다르지만, 저는 둘 다 레디스 마스터와 레플리카가 있다는 점이 눈에 띄었는데요.
주워들은 지식으로 명령과 조회의 책임 분리(Command and Query Responsibility Segregation, CQRS) 패턴이라는 녀석이 있습니다.

![read-side-and-write-side](https://user-images.githubusercontent.com/48385288/187067484-127d0efd-d0f5-46ae-90a1-b408722632c6.png)

말 그대로 어떠한 객체의 메서드는 명령(_command_) 형이거나 조회(_query_) 형 둘 중 하나라는 개념 정도까지만 알고 있는데,
책에서 소개하는 레디스 아키텍처가 이 CQRS 패턴도 같이 적용을 할 수 있는지? 혹은 CQRS 와는 상관이 없는 건지? 등이 궁금합니다.

**Comment by JoisFe** [comment, marked as answer]
성능 확장성 때문에 특히 분산 환경에서 CQRS 패턴을 사용하거나 고민하는 곳이 많다고 하네요
대놓고 레디스 공식사이트에서는 Redis Enterprise에서는 자신들의 제품을 이용해 CQRS 패턴을 구현하여 성능이나 확장성을 해결한 자료를 홍보도 하고 있네요
아래 참고자료 보시면 될 것 같습니다.

- [참고 자료](https://redis.com/solutions/use-cases/microservices/cqrs/)
